backend/oasst_backend/api/v1/tasks.py

```diff
@@ -276,27 +276,31 @@ async def tasks_interaction(
                 )
 
                 # here we store the text reply in the database
-                newMessage = pr.store_text_reply(
+                new_message = pr.store_text_reply(
                     text=interaction.text,
                     frontend_message_id=interaction.message_id,
                     user_frontend_message_id=interaction.user_message_id,
                 )
 
+                if not settings.DEBUG_SKIP_TOXICITY_CALCULATION:
+                    save_toxicity(interaction, pr, new_message)
```

Comment:
below for the embedding calculation, we implement this in a different way, i.e. we don't have a helper function in `hugging_face`, but delegate to `prompt_repository`. Is there a reason we cannot do it analogously here? I'm not saying one approach is better, but we should strive for consistency.

Reply:
Yes, well this helper function in huggingface is at the end using the `prompt_repository`. Just I divided the concerns to make the code more readable.

Reply:
For the creation of the rest api client of the embeddings calculation, I also refactored that code with this helper function (that used as well the pr underneath).

Reply:
yes, I saw that. the issue is, a function that is called `save_X`, I would expect to take `X` as one of the arguments and then save that. But here, the function is actually mainly responsible for getting X, while the actual saving part is delegated to the prompt repository. so the prompt repository itself is perfectly capable of "saving X" in a single line of code, and I don't think we need additional redirection for that.

the bigger problem is this: the current change is not abstraction, it's just indirection & moving code around, making things more complicated. if anything should be abstracted, it's the actual computation part, but that is as of now simply placed into a different spot, it's not made easier. you can easily see how this design is suboptimal in the new PR you opened (#578 ) because there, you actually have to duplicate the code of building the URL and doing the http request into both your new helper function and the endpoint, because you obviously don't want to save in the endpoint. isn't the purpose of such a helper function to prevent you from having to do exactly that? on top of that, you're now missing error handling in the new PR.

I just don't see the current change as having positive impact. it moves control flow unnecessarily from one file to the next without gaining anything. I'd leave the calling of HF and storing as it is done below for the embedding, that's much clearer. If we want to refactor anything (in a separate PR), we can make a helper to make it easier to call HF endpoints and then use that from both branches